src/lib/snark_work_lib/metrics.ml

```diff
@@ -33,6 +35,46 @@ let emit_single_metrics ~logger ~(single_spec : _ Single_spec.Poly.t)
       let transaction_type = Transaction_type.of_transaction transaction in
       ( match transaction_type with
       | `Zkapp_command ->
+          if pre_mesa_upgrade then (
+            let zkapp_command =
```

Comment:
@glyh what's the point of having this code?

I don't see value in attempting to support backwards compatibility for the metrics before the hard fork. I think it's alright to remove the old metrics if they are no longer used and we don't update metrics and just update metrics in the same way that we'll be doing that after the hard fork.

If you think that this should be different, please let me know which particular concerns are on the table.

UPD: saw the comment by Christian. Maybe we'll just not merge this PR before it's approved together with a follow-up PR that enables the feature.

Reply:
Agree with your point the supporting previous metrics will be a mess.

Reply:
Because we haven't deliver the upgrade in this PR yet.

Reply:
Yep, got that. Let me review it more thoroughly